### PR TITLE
Pass a Uri to package:http APIs

### PR DESCRIPTION
**Pull request by natebosch**
Prepare for https://github.com/dart-lang/http/issues/375

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+## 3.3.12-dev
+
```
Comment (devoncarew) [inline CHANGELOG.md:2]: Perhaps `- Updates to prepare for a new version of package:http`

**Review by devoncarew — COMMENTED** [review]
Thanks for the PR! In order to pass the CI, I believe you'll need to run `pub run build_runner build --delete-conflicting-outputs` locally, and add an entry to the changelog.

File: CHANGELOG.md (hunk 1 above)
Comment (natebosch) [reply CHANGELOG.md:2]: FWIW we don't need to publish this now, we just need it merged to google3. I typically avoid adding changelogs that don't impact the user for the sake of having something there, but I can add something.

Comment (devoncarew) [reply CHANGELOG.md:2]: > I typically avoid adding changelogs that don't impact the user for the sake of having something there, but I can add something.

Same here, but from a cursory look at the travis failures, it looks like we have some validation that there's a changelog entry for new versions.

**Review by devoncarew — APPROVED** [review]

**Pull request merged**